breadbox/breadbox/crud/dataset.py

```diff
@@ -477,6 +477,28 @@ def get_matrix_dataset_samples(
     return dataset_samples
 
 
+def get_matrix_dataset_given_ids(
+    db: SessionWithUser, dataset: Dataset, axis: str
+) -> List[str]:
+    assert_user_has_access_to_dataset(dataset, db.user)
+
+    if axis == "feature":
+        dimension_class = DatasetFeature
+    elif axis == "sample":
+        dimension_class = DatasetSample
+    else:
+        raise ValueError(f"Invalid axis: {axis}")
+
+    given_ids = (
```

Comment:
If this is a method worth improving, we could easily make it more efficient by create an index on dimension_class (dataset_id, given_id)